allensdk/brain_observatory/ecephys/behavior_ecephys_session.py

```diff
@@ -354,7 +405,7 @@ def from_nwb(
         instantiated `BehaviorEcephysSession`
         """
         kwargs['add_is_change_to_stimulus_presentations_table'] = False
-        behavior_session = VBNBehaviorSession.from_nwb(
+        behavior_session = cls.behavior_data_class().from_nwb(
```

Comment:
If you want this class to be specific to VBN then you should subclass the VBNBehaviorSession class rather than BehaviorSession and get rid of cls.behavior_data_class()

Reply:
This was my intent to honor your intent that this class be general.

I suspect you are right that BehaviorEcephysSession is going to be generally applicable to all Behavior-Ecephys datasets. I think we are learning that each experimental group wants to process the behavior data a little differently. My hope was that, if VBN2025 is different from VBN2022, we could just

```
class VBN2025BehaviorSession(BehaviorSession)
    .... _read_table class methods ...

class VBN20205BehaviorEcephysSession(BehaviorEcephysSesssion):

    @classmethod
    def behavior_data_class(cls):
            VBN2025BehaviorSession
```

And be done.

I do think that we have learned enough that we can split BehaviorSession and BehaviorEcephysSession into mixins that can be combined in new and unique ways to support different data releases. That will take too long and is out of scope fort his ticket, though.

Reply:
That makes sense, but I'm confused why you didn't subclass `VBNBehaviorSession`. Instead you added `behavior_data_class`. And it still subclasses `BehaviorSession`. I didn't fully appreciate this when I reviewed this change.

Reply:
I wasn't sure why BehaviorEcephysSession was a sub-class of BehaviorSession, given that from_json instantiated a "naked" BehaviorSession.

I initially tried to split out the user-facing parts of BehaviorSession into a mixin that both BehaviorSession and BehaviorEcephysSession could inherit from so that BehaviorEcephysSession could be its own thing, rather than a sub-class of BehaviorSession. Ultimately, that proved to be too much code for the PR at hand.

I think we should revisit this conversation. There's probably a more straightforward thing we can do with inheritance, rather than have BehaviorEcephysSession.from_json call BehaviorSession.from_json (or VBNBehaviorSession.from_json).

For now, I'd just like to merge this so we can start producing NWB files.